internal/controller/reinstall/reinstall.go

```diff
@@ -372,3 +379,228 @@ func (r *ReinstallHandler) finalizeReinstallRequest(
 	}
 	return nil
 }
+
+func (r *ReinstallHandler) ensureClusterInstanceSecretsHavePreservedLabel(
+	ctx context.Context,
+	log *zap.Logger,
+	clusterInstance *v1alpha1.ClusterInstance,
+) (reconcile.Result, error) {
+
+	preservationDataBackedup := findReinstallStatusCondition(clusterInstance, v1alpha1.ReinstallPreservationDataBackedup)
+	if preservationDataBackedup == nil {
+		return reconcile.Result{}, fmt.Errorf("condition %s has not been initialized",
+			v1alpha1.ReinstallPreservationDataBackedup)
+	}
+	if preservationDataBackedup.Reason != string(v1alpha1.Initialized) {
```

Comment:
Reading through the tests, this made me realize we're doing the same thing here. Changing behavior based on these conditions.

Do we not always want these resources to have this label? Why only set it when the reinstall process is initializing?

Reply:
I initially added this condition to handle cases where this function is re-run after the rendered manifests are deleted -at which point the BMC credential secret would also be deleted, leading to a `Not Found` error. However, since the ClusterInstance validator enforces that a BMC credential must be supplied (meaning it was likely already preserved), I removed these conditions. Instead, I added a check to simply ignore `Not Found` errors, making the execution more idempotent and reducing dependence on the resource's state.